### ISSUE TITLE
Support Nice and LimitNOFile in systemd

### ISSUE THREAD

**Issue by jordansissel** [jordansissel/pleaserun#120]
reference: https://discuss.elastic.co/t/rpm-packaging-problems-for-centos-6-7/63736/5

Currently pleaserun --nice is not reflected in systemd because the [template](https://github.com/jordansissel/pleaserun/blob/5c945397fe432cd9efa5e11dd2b9ce39b48ebe21/templates/systemd/default/program.service) doesn't support it.

Easy fix, I think.

**Comment by untergeek** [comment]
@jordansissel I looked into unit files and couldn't find a way to add it.  Is it just on the command-line?

**Comment by jordansissel** [comment]
@untergeek systemd's docs are a bit confusingly arranged, at least to me. I have to read 3 or 4 different manpages to find out everything you can set on a "service" unit. For my full brain dump:

* man systemd.service
* man systemd.exec
* man systemd.unit

The `systemd.exec` manpage is what details the `Nice=` and `LimitNOFILE=` settings. A `service` unit inherits the features  an `exec` unit provides, so you can have a service defined with a `Nice=` setting, etc.

**Comment by untergeek** [comment]
Of course. Inheritance. Super confusingly arranged in the documentation, I agree.

**Comment by jordansissel** [comment]
I've tested ed1564a by hand:

```
% sudo -E =ruby bin/pleaserun -p systemd --nice 15 --limit-open-files 12345 /usr/bin/sleep 600 
% sudo systemctl start sleep
% sudo systemctl status sleep
...
 Main PID: 15000 (sleep)

% ps -p 15000 -o %n
 NI
 15

% grep open /proc/15000/limits
Max open files            1024                 12345                files     
```

**Comment by untergeek** [comment]
@jordansissel Do we need another issue for upstart? Upstart on CentOS 6 doesn't seem to respect these values either:

```
$ cat /etc/logstash/startup.options  | egrep 'NICE|OPEN'
LS_OPEN_FILES=16384
LS_NICE=19
```

And the resulting upstart file:

```
$ cat /etc/init/logstash.conf | egrep -i 'NICE|OPEN|nofile'
#nice 19
#limit nice <softlimit> <hardlimit>
#limit nofile <softlimit> <hardlimit>
```

I'm not sure if the 19 is there because it's being carried over or not, but it's certainly not uncommenting it.  Also, `nofile` doesn't seem to be set.  Is that because it's a hard limit? Does `pleaserun` need to set both to the provided value?

**Comment by jordansissel** [comment]
those values are commented out probably because I was holding back when originally implementing upstart support. I can think of two reasons for me to have held back:

1) Upstart 0.6.5 is ancient and it was hard for me to find good documentation.
2) I wanted integration tests to verify behavior, but hadn't, at the time, made such tests good enough.

**Comment by jordansissel** [comment]
I'll have a go patching upstart as well.

**Comment by jordansissel** [comment]
a87bc8af28901493804c1ec6d31b80614173b6cd implements this for Upstart 0.6.5

**Comment by jordansissel** [comment]
pleaserun 0.0.25 published with this stuff.

**Comment by jordansissel** [comment]
pleaserun 0.0.26 published because I forgot to keep mustache at 0.99.x (for RUby 1.9 support)
